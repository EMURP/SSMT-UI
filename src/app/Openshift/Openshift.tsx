import * as React from 'react';
import { PageSection, Title }  from '@patternfly/react-core';
import {SimpleTable} from '@app/myTable/SimpleTable'
import { ChartPie, ChartThemeColor } from '@patternfly/react-charts';

//import {Table}from 'react-bootstrap-table';
//import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';

import { Card,  CardBody } from '@patternfly/react-core';
import {  } from '@patternfly/react-table';
import { Fetchdata } from '@app/fetchdata/fetchdata';

const horizontalFlex= {
  display: "flex",
  flexDirection: "row",
  marginRight: "30px",
  marginLeft: "30px"
} as React.CSSProperties

const padding= {

marginBottom: "30px",
marginRight: "30px",
marginLeft: "30px"

} as React.CSSProperties

const paddingChart= {

  marginBottom: "30px",
  marginRight: "30px",
  marginLeft: "30px",
  height: '275px',
  width: '300px'

} as React.CSSProperties

const Openshift: React.FunctionComponent<{}> = () => (


  <PageSection>


    <Title headingLevel="h1" size="lg">Openshift </Title>

  <div style={horizontalFlex}>

      <div style={padding}>
  
          <Card>
            <CardBody>Node Info Table </CardBody>
          </Card>

          <SimpleTable/>

      </div>

      <div style={padding}>

        <Card>
        <CardBody> Pod Info Table  </CardBody>
        </Card>

        <SimpleTable/>

    </div>
  
  </div>

  <div style={horizontalFlex}>

      <div style={padding}>
  
          <Card>
            <CardBody>Project Info Table </CardBody>
          </Card>

          <SimpleTable/>

      </div>

      <div style={paddingChart}>

     
      <ChartPie
      ariaDesc="Average number of Pods"
      ariaTitle="Pie chart example"
      constrainToVisibleArea={true}
      data={[{ x: 'Running', y: 35 }, { x: 'Failed', y: 55 }, { x: 'Terminated', y: 10 }]}
      height={275}
      labels={({ datum }) => `${datum.x}: ${datum.y}`}
      legendData={[{ name: 'Running: 35' }, { name: 'Failed: 55' }, { name: 'Terminated: 10' }]}
      legendPosition="bottom"
      padding={{
        bottom: 65,
        left: 20,
        right: 20,
        top: 20
      }}
      themeColor={ChartThemeColor.multiOrdered}
      width={400}
    />

    </div>
  
  </div>


  <div>
  <Fetchdata/>
  </div> 


  </PageSection>
)

export { Openshift };