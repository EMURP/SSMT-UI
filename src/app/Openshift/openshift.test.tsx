import React from 'react';
import { shallow } from 'enzyme';
import { Openshift } from './Openshift';

describe('Openshift tests', () => {

  test('should render default Openshift component', () => {
    const view = shallow(<Openshift />);
    expect(view).toMatchSnapshot();
  })

})