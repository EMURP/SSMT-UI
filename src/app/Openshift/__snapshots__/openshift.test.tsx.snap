// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Openshift tests should render default Openshift component 1`] = `
<Component>
  <Component
    headingLevel="h1"
    size="lg"
  >
    Openshift 
  </Component>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginLeft": "30px",
        "marginRight": "30px",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
        }
      }
    >
      <Component>
        <Component>
          Node Info Table 
        </Component>
      </Component>
      <SimpleTable />
    </div>
    <div
      style={
        Object {
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
        }
      }
    >
      <Component>
        <Component>
           Pod Info Table  
        </Component>
      </Component>
      <SimpleTable />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginLeft": "30px",
        "marginRight": "30px",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
        }
      }
    >
      <Component>
        <Component>
          Project Info Table 
        </Component>
      </Component>
      <SimpleTable />
    </div>
    <div
      style={
        Object {
          "height": "275px",
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
          "width": "300px",
        }
      }
    >
      <ChartPie
        ariaDesc="Average number of Pods"
        ariaTitle="Pie chart example"
        constrainToVisibleArea={true}
        data={
          Array [
            Object {
              "x": "Running",
              "y": 35,
            },
            Object {
              "x": "Failed",
              "y": 55,
            },
            Object {
              "x": "Terminated",
              "y": 10,
            },
          ]
        }
        height={275}
        labels={[Function]}
        legendData={
          Array [
            Object {
              "name": "Running: 35",
            },
            Object {
              "name": "Failed: 55",
            },
            Object {
              "name": "Terminated: 10",
            },
          ]
        }
        legendPosition="bottom"
        padding={
          Object {
            "bottom": 65,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        themeColor="multi-ordered"
        width={400}
      />
    </div>
  </div>
</Component>
`;
